plasmapy/plasma.py

```diff
@@ -0,0 +1,79 @@
+"""
+plasmapy.plasma
+===============
+
+Defines the core Plasma class used by PlasmaPy to represent plasma properties.
+"""
+
+import numpy as np
+import astropy.units as u
+
+mu0 = np.pi * 4.0e-7 * (u.newton / (u.amp**2))
+
+class Plasma():
+    """Core class for describing and calculating plasma parameters.
+
+
+
+    Attributes
+    ----------
+    x : `astropy.units.Quantity`
+        x-coordinates within the plasma domain. Equal to the `domain_x` input parameter.
+    y : `astropy.units.Quantity`
+        y-coordinates within the plasma domain. Equal to the `domain_y` input parameter.
+    z : `astropy.units.Quantity`
+        z-coordinates within the plasma domain. Equal to the `domain_z` input parameter.
+    grid : `astropy.units.Quantity`
+        (3, x, y, z) array containing the values of each coordinate at every point in the domain.
+    domain_shape : tuple
+        Shape of the plasma domain.
+    density : `astropy.units.Quantity`
+        (x, y, z) array of mass density at every point in the domain.
+    momentum : `astropy.units.Quantity`
+        (3, x, y, z) array of the momentum vector at every point in the domain.
+    pressure : `astropy.units.Quantity`
+        (x, y, z) array of pressure at every point in the domain.
+    magnetic_field : `astropy.units.Quantity`
+        (3, x, y, z) array of the magnetic field vector at every point in the domain.
+
+        
+    Parameters
+    ----------
+    domain_x : `astropy.units.Quantity`
+        1D array of x-coordinates for the plasma domain. Must have units convertable to length.
+    domain_y : `astropy.units.Quantity`
+        1D array of y-coordinates for the plasma domain. Must have units convertable to length.
+    domain_z : `astropy.units.Quantity`
+        1D array of z-coordinates for the plasma domain. Must have units convertable to length.
+    """
+    @u.quantity_input(domain_x=u.m, domain_y=u.m, domain_z=u.m)
+    def __init__(self, domain_x, domain_y, domain_z):
+        # Define domain sizes
+        self.x = domain_x
+        self.y = domain_y
+        self.z = domain_z
+
+        x, y, z = self.x.si.value, self.y.si.value, self.z.si.value
+        self.grid = np.meshgrid(x, y, z, indexing='ij') * u.m
+        self.domain_shape = (len(self.x), len(self.y), len(self.z))
+
+        # Initiate core plasma variables
+        self.density = np.zeros(self.domain_shape) * u.kg / u.m**3
+        self.momentum = np.zeros((3, *self.domain_shape)) * u.kg / (u.m**2 * u.s)
```

Comment:
I did but that was apparently a lie! It crashed when I tried to import the plasma class in python 2.x and when I tried with python 3.x, but I realize now the python 3.x crash was because I don't have numpy working for python 3.x ¯\_(ツ)_/¯

Sorry for the confusion.